File: frontend/src/types/prediction.ts
// Request for feature-based prediction
export interface PredictionRequest {
  name: string;
  age: number;
  gender: string;
  features: number[];
}

// Patient entity (from backend)
export interface Patient {
  id: number;
  name: string;
  age: number;
  gender: string;
}

// Prediction history record
export interface Prediction {
  id: number;
  filePath: string;
  originalFileName: string;

  // Patient info is returned as flat fields by the /api/predictions/latest endpoint
  // (avoids Hibernate proxy serialization issues with List<Prediction>)
  patientId?: number;
  patientName?: string;

  voiceConfidence?: number;
  handwritingConfidence?: number;
  gaitConfidence?: number;
  tremorConfidence?: number;
  visualConfidence?: number;
  reactionTimeMs?: number;

  finalPrediction: number;
  finalRisk: number;
  riskLevel?: "LOW" | "MEDIUM" | "HIGH";

  createdAt: string;
}

File: backend/src/main/java/com/neurosense/backend/entity/Prediction.java
package com.neurosense.backend.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Table(name = "predictions")

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder

public class Prediction {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String filePath;

    private String originalFileName;

    private Double voiceConfidence;

    private Double handwritingConfidence;

    private Double gaitConfidence;

    private Double tremorConfidence;

    private Double visualConfidence;

    private Integer reactionTimeMs;

    private Integer finalPrediction;

    private Double finalRisk;

    private LocalDateTime createdAt;

    // @JsonIgnore prevents Jackson from serializing the patient (avoids proxy crash).
    // FetchType.EAGER ensures pred.getPatient() is always available when the
    // controller manually extracts patient info — no LazyInitializationException.
    @JsonIgnore
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "patient_id")
    private Patient patient;

}